massa-wallet/src/lib.rs

```diff
@@ -24,27 +28,39 @@ pub struct Wallet {
     pub keys: Map<Address, (PublicKey, PrivateKey)>,
     /// Path to the file containing the private keys (not encrypted)
     pub wallet_path: PathBuf,
+    /// Password
+    pub password: String,
 }
 
 impl Wallet {
     /// Generates a new wallet initialized with the provided file content
-    pub fn new(path: PathBuf) -> Result<Wallet, WalletError> {
-        let keys = if path.is_file() {
-            serde_json::from_str::<Vec<PrivateKey>>(&std::fs::read_to_string(&path)?)?
+    pub fn new(path: PathBuf, password: String) -> Result<Wallet, WalletError> {
+        let content = &std::fs::read(&path)?[..];
```

Comment:
the read() will fail if the file does not exist and quit the whole function

Reply:
`Wallet::new(path, password)?` is only called in `massa-client/main.rs` right after the file was tested and created if it does not exist by `ask_password()`

Reply:
see comment above for an alternative way of doing it